runtime-watcher/internal/handler.go

```diff
@@ -152,24 +159,44 @@ func (h *Handler) validateResources(request *admissionv1.AdmissionRequest, modul
 			GroupVersionKind: request.Kind,
 			SubResource:      request.SubResource,
 		}
-		msg := h.sendRequestToKcpOnUpdate(resource, oldObject, object, moduleName)
-		return admissionMessage(msg)
+		changed, err := h.checkForChange(resource, oldObject, object)
+		if err != nil {
+			h.metrics.UpdateFailedKCPTotal()
```

Comment:
we have many reasons here to update this counter, can we introduce a label for this metric to indicate the reason?
e.g: (please also define a good reason name)
```
watcher_failed_kcp_total{reason="invalid subresource"}
watcher_failed_kcp_total{reason="resource owner name could not be determined"}
watcher_failed_kcp_total{reason="KCPAddress or KCPContract empty"}

```

Reply:
Sounds good added some reason values for that label. PTAL